src/mscorlib/corefx/System/Buffers/ArrayPool.cs

```diff
@@ -30,25 +40,7 @@ public abstract class ArrayPool<T>
         /// existing buffer being taken from the pool if an appropriate buffer is available or in a new 
         /// buffer being allocated if one is not available.
         /// </remarks>
-        public static ArrayPool<T> Shared => SharedPool.Value;
-
-        /// <summary>Stores a cached pool instance for T[].</summary>
-        /// <remarks>
-        /// Separated out into a nested class to enable lazy-initialization of the pool provided by
-        /// the runtime, only forced when Shared is used (and not when Create is called or when
-        /// other non-Shared accesses happen).
-        /// </remarks>
-        private static class SharedPool
-        {
-            /// <summary>Per-type cached pool.</summary>
-            /// <remarks>
-            /// byte[] and char[] are the most commonly pooled array types. For these we use a special pool type
-            /// optimized for very fast access speeds, at the expense of more memory consumption.
-            /// </remarks>
-            internal readonly static ArrayPool<T> Value =
-                typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
-                Create();
-        }
+        public static ArrayPool<T> Shared => s_sharedInstance;
```

Comment:
Nit: What you have is fine, but you could also use an auto-prop initializer, e.g.
```C#
public static ArrayPool<T> Shared { get; } =
    typeof(T) == typeof(byte) || typeof(T) == typeof(char) ? new TlsOverPerCoreLockedStacksArrayPool<T>() :
    Create();
```